Dockerfile_build

```diff
@@ -1,8 +1,12 @@
 # This file describes an image that is capable of building Flux.
 
-FROM golang:1.12
+FROM golang:1.12-alpine
 
-RUN apt-get update && apt-get install -y --no-install-recommends \
-		ruby \
-		ragel=6.9-1.1+b1 \
-	&& rm -rf /var/lib/apt/lists/*
+RUN apk add --no-cache \
+        bash \
+        git \
+        make \
+        gcc \
+        libc-dev \
```

Comment:
Why are gcc and libc-dev needed?

Reply:
I found that `go vet` seems to use CGO, so it needs them.  We wouldn't need them if we set `CGO_ENABLED=0` in the environment.  Do you have a preference?

Reply:
I did a little more digging on this issue, I think that `go vet` needs to build something for some reason, and through transitive dependencies something that needs CGO (I seem to recall that the `net` package uses CGO) gets compiled.

This is a recent change in Go 1.11 that seems not to have been fixed in Go 1.12:
https://github.com/golang/go/issues/26988

Maybe the best way forward would be to add CGO_ENABLED=0 to the environment, and if/when that Go bug gets fixed, we won't have any unneeded dependencies in the docker build.